source/slang/slang-ir-glsl-legalize.cpp

```diff
@@ -691,6 +691,9 @@ ScalarizedVal createSimpleGLSLGlobalVarying(
     {
         builder->addImportDecoration(globalParam, UnownedTerminatedStringSlice(systemValueInfo->name));
 
+        // Marks the instruction as using the import decorations name as is.
+        builder->addDecoration(globalParam, kIROp_UseLinkageNameDecoration);
```

Comment:
I don't think this new decoration should be required. *Every* instruction with linkage should use its mangled name as-is, and we should fix the underlying problem of why those aren't unique.